pennylane_lightning/core/src/lightning_qubit.toml

```diff
@@ -0,0 +1,78 @@
+schema = 1
+
+[device]
+name = "lightning.qubit"
+precision = ['float32', 'float64']
+
+[operations]
+observables = [
+	"NamedObs",
+	"HermitianObs",
+	"TensorObs",
+	"HamiltonianObs",
+]
+
+[[operations.gates]]
+full = [
+	"Identity",
+	"PauliX",
+	"PauliY",
+	"PauliZ",
+	"Hadamard",
+	"S",
+	"T",
+	"PhaseShift",
+	"RX",
+	"RY",
+	"RZ",
+	"Rot",
+	"CNOT",
+	"CY",
+	"CZ",
+	"SWAP",
+	"IsingXX",
+	"IsingXY",
+	"IsingYY",
+	"IsingZZ",
+	"ControlledPhaseShift",
+	"CRX",
+	"CRY",
+	"CRZ",
+	"CRot",
+	"CSWAP",
+	"Toffoli",
+	"MultiRZ",
+        "SingleExcitation",
+        "SingleExcitationMinus",
+        "SingleExcitationPlus",
+        "DoubleExcitation",
+        "DoubleExcitationMinus",
+        "DoubleExcitationPlus",
+]
+# QubitUnitary should never be here.
+# Gates in "matrix" are decomposed to QubitUnitary.
+# Having a QubitUnitary here would mean an infinite loop.
+matrix = [
```

Comment:
Good idea to make sure it never appears here, but I noticed it is missing from the list above. Should we not include it?

Reply:
No, this is the wrong interpretation. `OrbitalRotation`, `QFT` and others are not implemented via `NamedGate` but `applyMatrix`.

Reply:
Since there are an infinite number of gates, wouldn't it make more sense to only define "full" / native gates and all the others be "matrix" gates implicitly?

Reply:
"Adjoint(S)" is a gate? Or a meta operation that can modify a gate? And how does this map to our support of `adjoint` on the compiler?

Reply:
> Since there are an infinite number of gates, wouldn't it make more sense to only define "full" / native gates and all the others be "matrix" gates implicitly?

That would gates would never be the decomposed.

> Adjoint(S) is a gate? Or a meta operation that can modify a gate? And how does this map to our support of adjoint on the compiler?

That's a good question, I don't know how this is parsed exactly. Worth looking into.

Reply:
From offline discussion:

1. Also adding a set of decomposed gates.
2. `Adjoint(s)` is not a specialized kernel, it just sets a flag. I think since we are already differing from QIR spec by having this bool inverse flag it makes sense for `Adjoint(s)` to live in native supported gates right now in this toml file. I'll investigate how exactly it gets compiled and report back. If we were to keep the QIR spec, we should look into `__quantum__rt__callable_make_adjoint`.

Reply:
All gates are either "native", "decomp" or "matrix" https://github.com/PennyLaneAI/pennylane-lightning/pull/566/commits/6e2771d17f040f111573a2939d520d409a843c19

I kept some gates which are named op as "matrix" gates until we have the QIR QIS functions.